File: python/a2a/main.py
# /// script
# requires-python = ">=3.9"
# dependencies = ["promptrails>=0.9.0"]
# ///
#
# NOTE (API v2): promptrails 0.9.0 is not yet on PyPI. Until it publishes, run
# against the local sibling SDK from inside this folder, e.g.:
#   uv run --with-editable ../../../python-sdk main.py
# The pinned spec above reconciles automatically once 0.9.0 ships.
"""Agent-to-Agent (A2A) communication protocol."""

import os
from promptrails import PromptRails

client = PromptRails(api_key=os.environ["PROMPTRAILS_API_KEY"])

AGENT_ID = os.environ.get("AGENT_ID", "your-agent-id")

# Get agent card (A2A discovery)
card = client.a2a.get_agent_card(AGENT_ID)
print(f"Agent: {card.name}")
print(f"Description: {card.description}")
print(f"Skills: {card.skills}")

# Send a message (blocking — waits for completion)
task = client.a2a.send_message(
    AGENT_ID,
    message="Analyze the sales data for Q1 2026",
    blocking=True,
)
print(f"Task: {task.id}")
print(f"State: {task.status.state if task.status else 'unknown'}")
# Results come back as artifacts, each a list of parts.
for artifact in task.artifacts:
    for part in artifact.parts:
        print(f"Result: {part.text}")

# Send a non-blocking message
task = client.a2a.send_message(
    AGENT_ID,
    message="Generate a monthly report",
    blocking=False,
)
state = task.status.state if task.status else "unknown"
print(f"Task created: {task.id} (state: {state})")

# Poll for completion
task = client.a2a.get_task(task.id)
print(f"Task state: {task.status.state if task.status else 'unknown'}")

# List tasks
tasks = client.a2a.list_tasks(limit=5)
for t in tasks.data:
    print(f"  {t.id} — {t.status}")

# Cancel a running task
client.a2a.cancel_task(task.id)

client.close()

File: javascript/a2a/main.mjs
import { PromptRails } from "@promptrails/sdk";

const client = new PromptRails({
  apiKey: process.env.PROMPTRAILS_API_KEY,
});

const AGENT_ID = process.env.AGENT_ID || "your-agent-id";

// Get agent card (A2A discovery)
const card = await client.a2a.getAgentCard(AGENT_ID);
console.log(`Agent: ${card.name}`);
console.log(`Description: ${card.description}`);

// Send a blocking message
const task = await client.a2a.sendMessage(
  AGENT_ID,
  "Analyze the sales data for Q1 2026",
  { blocking: true },
);
console.log(`Task: ${task.id}`);
console.log(`State: ${task.status?.state}`);
// Results come back as artifacts, each a list of parts.
for (const artifact of task.artifacts ?? []) {
  for (const part of artifact.parts) {
    console.log(`Result: ${part.text}`);
  }
}

// Send a non-blocking message
const asyncTask = await client.a2a.sendMessage(
  AGENT_ID,
  "Generate a monthly report",
  { blocking: false },
);
console.log(`Task created: ${asyncTask.id} (state: ${asyncTask.status?.state})`);

// Poll for completion
const updated = await client.a2a.getTask(asyncTask.id);
console.log(`Task state: ${updated.status?.state}`);

// List tasks
const tasks = await client.a2a.listTasks({ limit: 5 });
for (const t of tasks.data) {
  console.log(`  ${t.id} — ${t.status?.state}`);
}

// Cancel a running task
await client.a2a.cancelTask(asyncTask.id);
